clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1220,6 +1223,7 @@ private void ensureValidRecordSize(int size) {
      * of <code>flush()</code> is that any previously sent record will have completed (e.g. <code>Future.isDone() == true</code>).
      * A request is considered completed when it is successfully acknowledged
      * according to the <code>acks</code> configuration you have specified or else it results in an error.
+     * Additionally, this method clears the last exception in the transaction and returns the transaction from the error state.
```

Comment:
> returns

Same question as above

Reply:
@C0urante It would be nice if you took a look.
Instead of implementing a custom handler for Producer (as suggested in KIP-1038), we came up with the idea of allowing the user to still commit transactions even if there were errors in `flush()`. 
Now we are thinking whether this change needs a KIP or not!